aunt-leahs-app/src/pages/newVolunteer.jsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import Head from "../components/header";
+import CustomButton from "../components/customButton";
+import TextInput from "../components/textInput";
+import constants from "../constants";
+import "../styles.css";
+import Tickbox from "../components/tickbox";
+
+class NewVolunteer extends React.Component {
+  /* 
+                
+  == Basic Information ==
+  First name / last name
+  Email address
+
+  == Address ==
+  Stress address
+  City / Province
+  Postal Code
+
+  == Emergency Contact ==
+  First name / last name
+  Relationship / phone number
+  Email address
+
+  Mailing list
+
+  Submit button just logs current state for now!
+  */
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      fname: "",
+      lname: "",
+      email: "",
+      address: "",
+      city: "",
+      province: "",
+      postal: "",
+      contactFName: "",
+      contactLName: "",
+      contactRelationship: "",
+      contactPhone: "",
+      contactEmail: "",
+      mailingList: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleTickbox = this.handleTickbox.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  handleChange(event, stateToChange) {
+    this.setState({ [stateToChange] : event.target.value });
```

Comment:
Rather than stateToChange, you might name the variable field, fieldName, or key to improve readability.